### DISCUSSION TITLE
Capability to Add Advanced Filters Similar to ignore_hosts in MITMProxy

### DISCUSSION THREAD

**Discussion by SeshaPhaniVV** [mitmproxy/mitmproxy#6848] [Intercept Everything!]
Hi Team,

First and foremost, thank you for developing and maintaining such an outstanding tool in MITMProxy.

I have a question regarding the implementation of advanced filtering mechanisms. For example, I am looking to add filters based on payload attributes such as size, headers etc., while maintaining TLS encryption integrity.

For instance, I would like to implement a filter that intercepts traffic if the payload size exceeds 10KB, and allows smaller payloads to pass through without interception.

Is there a way to achieve this within the current architecture of MITMProxy? If so, could you please provide guidance or reference materials on how to implement these filters without compromising the TLS encryption?

Thank you in advance for your assistance!

**Comment by mhils** [comment]
This is generally not possible due to the design of TLS. Headers and
payload size are not yet known when the decision to intercept needs to be
made. The client will only transmit this information once the handshake is
complete.
